### PR TITLE
`vars` within `dbt_project.yml` does not support rendering

### PR DESCRIPTION
**Pull request by dbeatty10**
[Preview](https://deploy-preview-3554--docs-getdbt-com.netlify.app/docs/build/project-variables#defining-variables-in-dbt_projectyml)

## What are you changing in this pull request and why?

`vars` within `dbt_project.yml` does not support rendering, but it has been requested https://github.com/dbt-labs/dbt-core/issues/3105 and https://github.com/dbt-labs/dbt-core/issues/4314.

In the meantime, we want to make it clear within our docs that this isn't supported.

See https://github.com/dbt-labs/dbt-core/issues/4314#issuecomment-979259394 for context.

## 🎩 

image

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.

### CODE REVIEW

File: website/docs/docs/build/project-variables.md [hunk 1]
```diff
@@ -17,6 +17,13 @@ Variables can be defined in two ways:
 
 ### Defining variables in `dbt_project.yml`
 
+
+:::info
+
+`vars` does not support rendering Jinja.
```
Comment (nghi-ly) [inline website/docs/docs/build/project-variables.md]:
```suggestion
`vars` doesn't support rendering Jinja.
```

**Review by nghi-ly — COMMENTED** [review]
looks good! left one minor nit

**Review by nghi-ly — APPROVED** [review]

**Comment by JoakimNilssonAtea** [comment]
This is not very clear. After all, I don't need the vars config object for rendering anything nor for parsing jinja. dbt handles this in the rest of the dbt_project.yml file so I don't need any special rendering functionality in vars.

The behavior is very unexpected, so it should be made very explicit. E.g. "Jinja is not supported in the vars config object, unlike in the rest of the dbt_project.yml file. All values will be interpreted literally."

**Comment by dbeatty10** [comment]
Thanks for that feedback @JoakimNilssonAtea. I tagged you in a draft PR here: https://github.com/dbt-labs/docs.getdbt.com/pull/3619#issuecomment-1607963139

**Pull request merged**